src/procedures/__tests__/ApproveErc20.ts

```diff
@@ -0,0 +1,111 @@
+import * as sinon from 'sinon';
+import { ImportMock, MockManager } from 'ts-mock-imports';
+import { SinonStub } from 'sinon';
+import * as contextObject from '../../Context';
+import * as polymathBaseObject from '../../PolymathBase';
+import * as contractWrappersObject from '@polymathnetwork/contract-wrappers';
+import { ApproveErc20 } from '../../procedures/ApproveErc20';
+import { Procedure } from '~/procedures/Procedure';
+import BigNumber from 'bignumber.js';
+import { Wallet } from '~/Wallet';
+
+const params1 = {
+  amount: new BigNumber(1),
+  spender: '0x1',
+  owner: '0x3',
+};
+
+const params2 = {
+  amount: new BigNumber(1),
+  spender: '0x1',
+  owner: '0x3',
+  tokenAddress: '0x2222222222222222222222222222222222222222',
+};
+
+describe('ApproveErc20', () => {
+  let target: ApproveErc20;
+  let contextMock: MockManager<contextObject.Context>;
+  let wrappersMock: MockManager<polymathBaseObject.PolymathBase>;
+  let erc20Mock: MockManager<contractWrappersObject.ERC20>;
+  let polyTokenMock: MockManager<contractWrappersObject.PolyToken>;
+
+  let checkPolyAllowanceStub: SinonStub<any, any>;
+  let checkPolyAddressStub: SinonStub<any, any>;
+  let checkPolyBalanceStub: SinonStub<any, any>;
+
+  let checkErc20AllowanceStub: SinonStub<any, any>;
+  let checkErc20AddressStub: SinonStub<any, any>;
+  let checkErc20BalanceStub: SinonStub<any, any>;
+
+  beforeAll(() => {
+    // Mock the context and wrappers, including currentWallet and balanceOf to test ApproveErc20
+    contextMock = ImportMock.mockClass(contextObject, 'Context');
+    wrappersMock = ImportMock.mockClass(polymathBaseObject, 'PolymathBase');
+    erc20Mock = ImportMock.mockClass(contractWrappersObject, 'ERC20');
+    polyTokenMock = ImportMock.mockClass(contractWrappersObject, 'PolyToken');
+
+    // Setup poly token
+    checkPolyBalanceStub = polyTokenMock.mock('balanceOf', Promise.resolve(new BigNumber(2)));
+    checkPolyAddressStub = polyTokenMock.mock('address', Promise.resolve(params1.spender));
+    checkPolyAllowanceStub = polyTokenMock.mock('allowance', Promise.resolve(new BigNumber(0)));
+
+    contextMock.set('contractWrappers', wrappersMock.getMockInstance());
+    wrappersMock.set('polyToken', polyTokenMock.getMockInstance());
+    wrappersMock.mock('isTestnet', Promise.resolve(false));
+    const ownerPromise = new Promise<string>((resolve, reject) => {
+      resolve();
+    });
+    contextMock.set('currentWallet', new Wallet({ address: () => ownerPromise }));
```

Comment:
This can be rewritten using `Promise.resolve(ownerAddress)` instead of creating a new Promise object.

Also, why is this resolving to nothing? Shouldn't it be a string?